projects/kit/src/lib/masks/date/utils/parse-date.ts

```diff
@@ -0,0 +1,17 @@
+import {DEFAULT_MAX_DATE, DEFAULT_MIN_DATE} from '../../../constants';
+import {parseDateString, segmentsToDate} from '../../../utils';
+import type {MaskitoDateParams} from '../date-mask';
+
+export function maskitoParseDate(
+    value: string,
+    {mode, min, max}: MaskitoDateParams,
+): Date {
+    const dateSegments = parseDateString(value, mode);
+
+    const parsedDateTime = segmentsToDate(dateSegments).getTime();
+
+    const minTime = (min ?? DEFAULT_MIN_DATE).getTime();
+    const maxTime = (max ?? DEFAULT_MAX_DATE).getTime();
+
+    return new Date(Math.min(Math.max(parsedDateTime, minTime), maxTime));
```

Comment:
For this case we already have `clamp`-utility (it works with `Date` objects too)

https://github.com/taiga-family/maskito/blob/5c578b32a3a73662fd6031f88bc89f6e3e682edc/projects/kit/src/lib/utils/clamp.ts#L8-L11